src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1072,7 +1072,7 @@ public void OpenDocumentCore(DocumentId documentId, bool activate = true)
                 throw new ArgumentNullException(nameof(documentId));
             }
 
-            if (!ThreadHelper.JoinableTaskContext.IsOnMainThread)
+            if (!_foregroundObject.ThreadingContext.HasMainThread)
```

Comment:
⚠️ This completely changed the meaning of the assertion. Are you sure this was intentional?

Reply:
Whoops good catch!